src/objects/zcl_abapgit_object_iobj.clas.abap

```diff
@@ -0,0 +1,207 @@
+CLASS zcl_abapgit_object_iobj DEFINITION PUBLIC INHERITING FROM zcl_abapgit_objects_super FINAL.
+
+  PUBLIC SECTION.
+    INTERFACES zif_abapgit_object.
+    ALIASES mo_files FOR zif_abapgit_object~mo_files.
+
+  PROTECTED SECTION.
+  PRIVATE SECTION.
+ENDCLASS.
+
+
+
+CLASS zcl_abapgit_object_iobj IMPLEMENTATION.
+
+
+  METHOD zif_abapgit_object~changed_by.
+
+    DATA: ls_viobj TYPE rsd_s_viobj,
+          lv_objna TYPE rsd_iobjnm.
+
+    lv_objna = ms_item-obj_name.
+
+    CALL FUNCTION 'RSD_IOBJ_GET'
+      EXPORTING
+        i_iobjnm  = lv_objna
+        i_objvers = 'A'
+      IMPORTING
+        e_s_viobj = ls_viobj.
+
+    rv_user = ls_viobj-tstpnm.
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~delete.
+
+    DATA: lt_iobjname     TYPE rsd_t_c30,
+          lv_object       TYPE string,
+          lv_object_class TYPE string.
+
+    APPEND ms_item-obj_name TO lt_iobjname.
+
+    CALL FUNCTION 'RSDG_IOBJ_MULTI_DELETE'
+      EXPORTING
+        i_t_iobjnm = lt_iobjname.
+
+    IF sy-subrc <> 0.
+      zcx_abapgit_exception=>raise( |Error when deleting infoObject | ).
+    ENDIF.
+
+    lv_object_class = ms_item-obj_type.
+    lv_object       = ms_item-obj_name.
+
+    CALL FUNCTION 'RS_CORR_INSERT'
+      EXPORTING
+        object              = lv_object
+        object_class        = lv_object_class
+        master_language     = mv_language
+        global_lock         = abap_true
+        mode                = 'D'
+        suppress_dialog     = abap_true
+      EXCEPTIONS
+        cancelled           = 1
+        permission_failure  = 2
+        unknown_objectclass = 3
+        OTHERS              = 4.
+    IF sy-subrc <> 0.
+      zcx_abapgit_exception=>raise_t100( ).
+    ENDIF.
```

Comment:
Suggest to use `rs_metadata-delete_tadir = abap_true.` in `zif_abapgit_object~get_metadata` instead.

Reply:
Changed as requested

Reply:
You can then delete the above marked code

Reply:
@christianguenter2 , are we shure about that? Maybe I misunderstood something, but after comment this part of code I can't uninstall my repository and add deletion to transport, please check. Maybe I can create some kind of check before the exception? Sorry for this babysitting of me guys, but ABAPGit from creator point of view quite new for me.
![2019-11-19 22 00 54](https://user-images.githubusercontent.com/11619019/69186462-d7734d80-0b18-11ea-8f7c-5c7c92601b5d.gif)

Reply:
No I'm not sure, I just assumed that. If it works I'm fine with that. Sorry for confusion.
Let me know when I can review again.